prov/tcp/src/tcpx_ep.c

```diff
@@ -170,7 +170,13 @@ static ssize_t tcpx_sendmsg(struct fid_ep *ep, const struct fi_msg *msg,
 	send_entry->done_len = 0;
 
 	fastlock_acquire(&tcpx_ep->queue_lock);
-	dlist_insert_tail(&send_entry->entry, &tcpx_ep->tx_queue);
+	if (dlist_empty(&tcpx_ep->tx_queue)) {
+		/* fast path for small messages */
+		dlist_insert_tail(&send_entry->entry, &tcpx_ep->tx_queue);
+		process_tx_pe_entry(send_entry);
+	} else {
+		dlist_insert_tail(&send_entry->entry, &tcpx_ep->tx_queue);
+	}
```

Comment:
The 'fast' path here is based on the tx_queue being empty, and not on message size.